### PR TITLE
Update XeTile.md with pack/unpack, inner_block, order

### PR DESCRIPTION
**Pull request by Jianhui-Li**
Please review these guidelines to help with the review process:
- [ ] Have you provided a meaningful PR description?
- [ ] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?
- [ ] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

File: docs/rfcs/XeTile.md [hunk 1]
```diff
@@ -25,60 +25,113 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
-  %block = XeTile.init_tile %base_addr, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
+  %tile0 = XeTile.init_tile %base_addr, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
      i64 into tile<8x16xbf16>
 ```
+
+`init_tile` with an `order` to access the base matrix. The `order` attribute describes the order of the tile elements stored in the memory. "0" indicates the fastest-changing dimension. So if the base matrix is stored as row-major, the order is specified as [1, 0]. If the base matrix is stored as column major, the order is specified as [0, 1]. The default is row-major. The output tile carries the `order` attribute in its attribute set. 
+
+```mlir
+  #tile_attr = #xetile.tile_ttr<order = [0, 1]>
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2] {order = [0, 1]}:
```
Comment (charithaintc) [inline docs/rfcs/XeTile.md:55]: remove `order` from the op

File: docs/rfcs/XeTile.md [hunk 2]
```diff
@@ -25,60 +25,113 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
+     memref<128x128xbf16> into tile<64x32xbf16, #tile_attr>
+```
+
+`init_tile` with an `inner_block` for 2D block access of the base matrix. The `inner_blocks` attribute describes the block size for each memory load and store operation when the tile is being loaded. The block size for load may be larger than the block size for MMA operation. The output tile carries the `inner_block` attribute in its attribute set. 
+
+```mlir
+  #tile_attr = #xetile.tile_ttr<inner_blocks=[16,16]>
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2] {inner_blocks=[16,16]}:
```
Comment (charithaintc) [inline docs/rfcs/XeTile.md:63]: remove `inner_block` from the op

File: docs/rfcs/XeTile.md [hunk 3]
```diff
@@ -25,60 +25,113 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
-`load_tile` loads a tile to a vector, which could be backed by a register region.
+`load_tile` loads a tile to a 2D vector, which could be backed by a register region.
 ```mlir
   %vector_a = XeTile.load_tile %tile_a :
      tile<64x64xbf16> into vector<64x64xb16>
 ```
 Attribute `transpose` specifies the dimensions being transposed along the load. It is commonly used for the GEMM on the backward path of DNN model, where one of input matrices needs to be transposed for matmul operation.
 ```mlir
-  %vector_a = XeTile.load_tile %tile_a {transpose = [1, 0]} :
+  %vector_a = XeTile.load_tile %tile_a {transpose = [0, 1]} :
```
Comment (charithaintc) [inline docs/rfcs/XeTile.md:76]: are we keeping the transpose?

File: docs/rfcs/XeTile.md [hunk 4]
```diff
@@ -91,17 +144,32 @@ A `tile_mma` variant without vector_c initialization.
 		tile<64x64xbf16>, index, index into tile <64x64xbf16>
 ```
 
+`tile_pack` packs a 2D vector, representing the loaded value from 2D tile, to a 4D vector with an inner block size. The 4D vector was introduced to support blocking to fit the hardware matrix operation sizes.  The blocking follows an implicit rule: dim[0] is split to dim[0] and dim[2], and dim[1] is split to dim[1] and dim[3]. The dim[2] and dim[3] of result 4D vector must be same as the size of `inner_blocks` attribute.
```
Comment (charithaintc) [inline docs/rfcs/XeTile.md:147]: .. an implicit rule: `out_dim[0] = in_dim[0]/inner_blocks[0] , out_dim[1] = in_dim[1]/inner_blocks[1], out_dim[2] = inner_blocks[0], and out_dim[3] = inner_blocks[1]`

**Review by charithaintc — APPROVED** [review]
Approve with comments

File: docs/rfcs/XeTile.md (hunk 3 above)
Comment (Jianhui-Li) [reply docs/rfcs/XeTile.md:76]: I am thinking removing it.  Need more discussions.

Comment (Jianhui-Li) [reply docs/rfcs/XeTile.md:76]: I think the user may use both programming models: matmul (row_major, col_major), and matmul(row_major, row_maj^T). 
Below is one example for using row_major^T. However, when user uses this example, user will do the transpose explicitly and not combine with load. So it is XeTile's internal optimization which can take load and transpose and optimized to load with transpose at XeGPU level. 
https://github.com/openai/triton/blob/969c5bb1b828518c672376cf653f10ae13a0cfb4/test/TritonGPU/loop-pipeline.mlir#L552

File: docs/rfcs/XeTile.md [hunk 5]
```diff
@@ -25,60 +25,109 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
 ```mlir
-  %block = XeTile.init_tile %base_addr, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
+  %tile0 = XeTile.init_tile %base_addr, [%tile_offset:2], [%base_shape:2], [%base_strides:2]:
      i64 into tile<8x16xbf16>
 ```
+
+`init_tile` with an `order` to access the base matrix. The `order` attribute describes the order of the tile elements stored in the memory. "0" indicates the fastest-changing dimension. So if the base matrix is stored as row-major, the order is specified as [1, 0]. If the base matrix is stored as column major, the order is specified as [0, 1]. The default is row-major. The output tile carries the `order` attribute in its attribute set. 
+
+```mlir
+  #tile_attr = #xetile.tile_ttr<order = [0, 1]>
```
Comment (silee2) [inline docs/rfcs/XeTile.md:54]: I think this is a typo
tile_ttr -> tile_attr

File: docs/rfcs/XeTile.md [hunk 6]
```diff
@@ -25,60 +25,109 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
+  %tile0 = XeTile.init_tile %base_memref, [%tile_offset:2]:
+     memref<128x128xbf16> into tile<64x32xbf16, #tile_attr>
+```
+
+`init_tile` with an `inner_block` for 2D block access of the base matrix. The `inner_blocks` attribute describes the block size for each memory load and store operation when the tile is being loaded. The block size for load may be larger than the block size for MMA operation. The output tile carries the `inner_block` attribute in its attribute set. 
+
+```mlir
+  #tile_attr = #xetile.tile_ttr<inner_blocks=[16,16]>
```
Comment (silee2) [inline docs/rfcs/XeTile.md:62]: Typo?
tile_ttr -> tile_attr

File: docs/rfcs/XeTile.md [hunk 7]
```diff
@@ -25,60 +25,109 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
-     tile<32x64xbf16> into vector<64x32xbf16>
-```
+
 Attribute `padding` specifies the padding value for the out-of-boundary access. The default value is zero.  
 ```mlir
   %vector_a = XeTile.load_tile %tile_a {padding = 1.0} :
      tile<64x64xbf16> into vector<64x64xb16>
 ```
-`load_tile` need to be used together with the tile_mma.
+`load_tile` needs to be used together with the tile_mma.
```
Comment (silee2) [inline docs/rfcs/XeTile.md:80]: "together" sounds redundant.
Just "needs to be used with tile_mma." should be fine.

File: docs/rfcs/XeTile.md [hunk 8]
```diff
@@ -25,60 +25,109 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
+
+`load_tile` loads a tile with an `order` attribute. The `order` attribute only affects the physical memory address calculation for the tile, however, it doesn't change the logical representation of loading a 2D tile to a 2D vector. There is no need to reverse the order of vector length at the XeTile level regardless of the `order` attribute value.
```
Comment (silee2) [inline docs/rfcs/XeTile.md:82]: Just a suggestion but

"the physical memory address calculation for the tile, however, it"
=> "indexing order and"

File: docs/rfcs/XeTile.md [hunk 9]
```diff
@@ -25,60 +25,109 @@ XeTile provides a middle-level abstraction for matmul operation and sits between
...
+  %vector_a = XeTile.load_tile %tile_a :
+     tile<64x32xbf16> into vector<4x2x16x16xb16, #tile_attr>
+```
 
 `store_tile` stores a vector to memory. Transpose and padding attributes are not supported.
 ```mlir  
   XeTile.store_tile %tile_a, %vector_a :
    vector<64x64xbf16> into tile<64x64xbf16>
 ```
+`store_tile` loads a tile with an `order` attribute. With the `order` attribute, there is no need to reorder the 2D vector dimension before saving it to a column-major matrix storage.
```
Comment (silee2) [inline docs/rfcs/XeTile.md:101]: "loads a tile" => "stores a tile"
"with an" =>  "according to the"
"dimension before saving it to a column-major matrix storage."
"dimensions. The op does the reordering."

File: docs/rfcs/XeTile.md [hunk 10]
```diff
@@ -91,17 +140,37 @@ A `tile_mma` variant without vector_c initialization.
 		tile<64x64xbf16>, index, index into tile <64x64xbf16>
 ```
 
+`tile_pack` packs a 2D vector, representing the loaded value from 2D tile, to a 4D vector with an inner block size. The 4D vector was introduced to support blocking to fit the hardware matrix operation sizes.  The blocking follows an implicit rule: out_dim[0] = in_dim[0]/inner_blocks[0] , out_dim[1] = in_dim[1]/inner_blocks[1], out_dim[2] = inner_blocks[0], and out_dim[3] = inner_blocks[1]. The dim[2] and dim[3] of result 4D vector must be same as the size of `inner_blocks` attribute. 
+
+```mlir
+  %0 = XeTile.tile_pack %1 inner_blocks = [16, 16]
+    : vector<64x32xf32> -> vector<4x2x16x16xf32>
+```
+`tile_unpack` unpacks a 4D blocked vector back to original 2D unpacked 2D vector.
```
Comment (silee2) [inline docs/rfcs/XeTile.md:149]: "original 2D unpacked 2D vector."
=> "original unpacked 2D vector

**Review by silee2 — APPROVED** [review]
LGTM

**Pull request merged**